### PR TITLE
Regionmanager polished

### PR DESCRIPTION
**Pull request by Zemurin**
Dropped eu4tovic2's regoinmapper for the streamlined ck3toeu4 regionmanager, cleaned it up a bit.

Fully tested, no brains required.

### CODE REVIEW

File: EU4toV3/Source/EU4World/RegionManager/RegionManager.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	void registerSuperRegionKeys();
+	void registerRegionKeys();
+	void linkSuperRegions();
+	void linkRegions();
+
+	std::map<std::string, std::shared_ptr<Area>> areas;
+	std::map<std::string, std::shared_ptr<Region>> regions;
+	std::map<std::string, std::shared_ptr<SuperRegion>> superRegions;
+};
+} // namespace mappers
```
Comment (IhateTrains) [inline EU4toV3/Source/EU4World/RegionManager/RegionManager.h:34]: Namespace is EU4.

File: EU4toV3/Source/EU4World/RegionManager/Region.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void EU4::Region::registerKeys()
+{
+	registerKeyword("areas", [this](std::istream& theStream) {
+		for (const auto& name: commonItems::stringList(theStream).getStrings())
```
Comment (IhateTrains) [inline EU4toV3/Source/EU4World/RegionManager/Region.cpp:16]:
```suggestion
		for (const auto& name: commonItems::getStrings(theStream))
```

File: EU4toV3/Source/EU4World/RegionManager/Area.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+{
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void EU4::Area::registerKeys()
+{
+	registerKeyword("color", commonItems::ignoreItem);
+	registerRegex("[0-9]+", [this](const std::string& number, std::istream& theStream) {
```
Comment (IhateTrains) [inline EU4toV3/Source/EU4World/RegionManager/Area.cpp:15]:
```suggestion
	registerRegex(commonItems::integerRegex, [this](const std::string& number, std::istream& theStream) {
```

**Review by IhateTrains — APPROVED** [review]

File: EU4toV3Tests/EU4WorldTests/RegionManagerTests/SuperRegionTests.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	EXPECT_TRUE(superRegion.getRegions().empty());
+}
+
+TEST(Mappers_SuperRegionTests, RegionsCanBeLoaded)
+{
+	const std::vector<std::string> input = {"region1", "region2"};
+	const EU4::SuperRegion superRegion(input);
+
+	EXPECT_TRUE(superRegion.getRegions().contains("region1"));
```
Comment (Idhrendur) [inline EU4toV3Tests/EU4WorldTests/RegionManagerTests/SuperRegionTests.cpp:19]: For completion's sake, also expect region2

**Review by Idhrendur — APPROVED** [review]

File: EU4toV3Tests/EU4WorldTests/RegionManagerTests/RegionTests.cpp [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	EXPECT_TRUE(region.getAreas().contains("test_area"));
+}
+
+TEST(Mappers_RegionTests, multipleAreasCanBeLoaded)
+{
+	std::stringstream input;
+	input << "areas = { test_area area2 area3 } \n";
+	const EU4::Region region(input);
+
+	EXPECT_EQ(3, region.getAreas().size());
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply EU4toV3Tests/EU4WorldTests/RegionManagerTests/RegionTests.cpp:30]: Unfortunately no, can't test maps like that. I tried messing with extracting keys through views but that also exploded.

[the remark this answers is not in the document]
Comment (Zemurin) [reply EU4toV3Tests/EU4WorldTests/RegionManagerTests/RegionTests.cpp:30]: image

File: EU4toV3Tests/EU4WorldTests/RegionManagerTests/RegionTests.cpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+TEST(Mappers_RegionTests, areaCanBeLoaded)
+{
+	std::stringstream input;
+	input << "areas = { test_area } \n";
+	const EU4::Region region(input);
+
+	EXPECT_FALSE(region.getAreas().empty());
+	EXPECT_TRUE(region.getAreas().contains("test_area"));
```
[the remark this answers is not in the document]
Comment (Zemurin) [reply EU4toV3Tests/EU4WorldTests/RegionManagerTests/RegionTests.cpp:21]: same issue, one or multiple pairs, it's all red.

**Pull request merged**